compiler/rustc_codegen_cranelift/src/unsize.rs

```diff
@@ -58,24 +86,29 @@ fn unsize_thin_ptr<'tcx>(
         (&ty::Adt(def_a, _), &ty::Adt(def_b, _)) => {
             assert_eq!(def_a, def_b);
 
-            let mut result = None;
-            for i in 0..src_layout.fields.count() {
-                let src_f = src_layout.field(fx, i);
-                assert_eq!(src_layout.fields.offset(i).bytes(), 0);
-                assert_eq!(dst_layout.fields.offset(i).bytes(), 0);
-                if src_f.is_zst() {
-                    continue;
+            if src_layout == dst_layout {
+                assert!(old_info.is_some());
+                (src, old_info.unwrap())
```

Comment:
Can you use a `return` here to avoid indenting the rest of the code?

Reply:
Done